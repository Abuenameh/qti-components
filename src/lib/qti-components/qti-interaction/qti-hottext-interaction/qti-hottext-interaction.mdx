import { Canvas, Meta } from '@storybook/blocks';

import * as Stories from './qti-hottext-interaction.stories';

<Meta title="qti-interactions/qti-hottext-interaction" />

# qti-hottext-interaction

<Canvas of={Stories.Default} />

<Canvas of={Stories.Button} />
