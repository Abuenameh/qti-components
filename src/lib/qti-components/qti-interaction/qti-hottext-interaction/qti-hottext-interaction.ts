import { html } from 'lit';
import { Choices } from '../internal/choices/choices';

export class QtiHottextInteraction extends Choices {
  override connectedCallback() {
    super.connectedCallback();
    this.setAttribute('qti-hottext-interaction', '');
  }

  override render = () => html`<slot></slot>`;
}

customElements.define('qti-hottext-interaction', QtiHottextInteraction);
