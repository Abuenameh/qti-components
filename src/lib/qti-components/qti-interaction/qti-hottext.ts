import { customElement } from 'lit/decorators.js';
import { QtiChoice } from './internal/choice/qti-choice';
import { html } from 'lit';

@customElement('qti-hottext')
export class QtiHottext extends QtiChoice {
  override render() {
    return html`<div part="ch"><div part="cha"></div></div>
      <slot></slot> `;
  }
}

declare global {
  interface HTMLElementTagNameMap {
    'qti-hottext': QtiHottext;
  }
}
