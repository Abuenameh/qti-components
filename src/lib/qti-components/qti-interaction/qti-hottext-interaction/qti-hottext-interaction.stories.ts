import { action } from '@storybook/addon-actions';
import { html } from 'lit';

import { ifDefined } from 'lit/directives/if-defined.js';
import '../qti-hottext';
import './qti-hottext-interaction';

const responses = ['[]', '["a"]', '"b"', '["d","e"]'];

export default {
  component: 'qti-hottext-interaction',
  parameters: {
    actions: {
      handles: ['saveResponse']
    },
    response: {
      control: { type: 'select', options: responses }
    }
  }
};

export const Default = {
  render: args => html`
    <qti-hottext-interaction
      @qti-interaction-response="${e => {
        action(JSON.stringify(e.detail))();
      }}"
      .response=${args.response}
      class=${ifDefined(args.classes ? args.classes.join(' ') : undefined)}
      max-choices=${args['max-choices']}
      response-identifier="RESPONSE"
    >
      <p>
        Sponsors of the Olympic Games
        <qti-hottext identifier="A">who bought</qti-hottext> advertising time on United States television
        <qti-hottext identifier="B">includes</qti-hottext>
        <qti-hottext identifier="C">at least</qti-hottext> a dozen international firms
        <qti-hottext identifier="D">whose</qti-hottext> names are familiar to American consumers.
        <qti-hottext identifier="E">No error.</qti-hottext>
      </p>
    </qti-hottext-interaction>
  `,

  args: {
    maxChoices: 1,
    classes: ''
  }
};

export const Button = {
  render: Default.render,

  args: {
    maxChoices: 1,
    classes: ['qti-input-control-hidden']
  }
};

export const Hidden = {
  render: Default.render,

  args: {
    maxChoices: 1,
    classes: ['qti-unselected-hidden']
  }
};
