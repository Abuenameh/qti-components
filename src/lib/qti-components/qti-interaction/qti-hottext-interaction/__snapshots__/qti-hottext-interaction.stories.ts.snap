// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/qti-components/qti-interaction/qti-hottext-interaction Button smoke-test 1`] = `
<div id="root-inner">
  <qti-hottext-interaction response-identifier="RESPONSE"
                           class="qti-input-control-hidden"
                           max-choices
                           qti-hottext-interaction
  >
    <p>
      Sponsors of the Olympic Games
      <qti-hottext identifier="A"
                   role="checkbox"
                   tabindex="0"
                   aria-disabled="false"
                   aria-readonly="false"
                   aria-checked="false"
      >
        who bought
      </qti-hottext>
      advertising time on United States television
      <qti-hottext identifier="B"
                   role="checkbox"
                   tabindex="0"
                   aria-disabled="false"
                   aria-readonly="false"
                   aria-checked="false"
      >
        includes
      </qti-hottext>
      <qti-hottext identifier="C"
                   role="checkbox"
                   tabindex="0"
                   aria-disabled="false"
                   aria-readonly="false"
                   aria-checked="false"
      >
        at least
      </qti-hottext>
      a dozen international firms
      <qti-hottext identifier="D"
                   role="checkbox"
                   tabindex="0"
                   aria-disabled="false"
                   aria-readonly="false"
                   aria-checked="false"
      >
        whose
      </qti-hottext>
      names are familiar to American consumers.
      <qti-hottext identifier="E"
                   role="checkbox"
                   tabindex="0"
                   aria-disabled="false"
                   aria-readonly="false"
                   aria-checked="false"
      >
        No error.
      </qti-hottext>
    </p>
  </qti-hottext-interaction>
</div>
`;

exports[`lib/qti-components/qti-interaction/qti-hottext-interaction Default smoke-test 1`] = `
<div id="root-inner">
  <qti-hottext-interaction response-identifier="RESPONSE"
                           max-choices
                           qti-hottext-interaction
  >
    <p>
      Sponsors of the Olympic Games
      <qti-hottext identifier="A"
                   role="checkbox"
                   tabindex="0"
                   aria-disabled="false"
                   aria-readonly="false"
                   aria-checked="false"
      >
        who bought
      </qti-hottext>
      advertising time on United States television
      <qti-hottext identifier="B"
                   role="checkbox"
                   tabindex="0"
                   aria-disabled="false"
                   aria-readonly="false"
                   aria-checked="false"
      >
        includes
      </qti-hottext>
      <qti-hottext identifier="C"
                   role="checkbox"
                   tabindex="0"
                   aria-disabled="false"
                   aria-readonly="false"
                   aria-checked="false"
      >
        at least
      </qti-hottext>
      a dozen international firms
      <qti-hottext identifier="D"
                   role="checkbox"
                   tabindex="0"
                   aria-disabled="false"
                   aria-readonly="false"
                   aria-checked="false"
      >
        whose
      </qti-hottext>
      names are familiar to American consumers.
      <qti-hottext identifier="E"
                   role="checkbox"
                   tabindex="0"
                   aria-disabled="false"
                   aria-readonly="false"
                   aria-checked="false"
      >
        No error.
      </qti-hottext>
    </p>
  </qti-hottext-interaction>
</div>
`;
